inc/event_listeners.h

```diff
@@ -0,0 +1,49 @@
+#ifndef EVENT_LISTENERS_H
+#define EVENT_LISTENERS_H
+
+#include <vector>
+#include <string>
+#include <iostream>
+#include <tuple>
+
+#include "events.h"
+
+#include "listeners/heartbeat.h"
+
+inline std::vector<std::string> listener_names = {"Heartbeat"};

```

Comment:
```suggestion
template <typename>
constexpr auto make_listener_names();

template <typename... Listeners>
constexpr auto make_listener_names<std::tuple<Listeners...>>() {
    return std::array<char const*, sizeof...(Listeners)>{{Listeners::cli_key...}};
}

constexpr inline auto listener_names = make_listener_names<decltype(listeners)>();

```
For your consideration: You may have an opportunity here to move the CLI key into the class (here, I've called it `Heartbeat::cli_key`). This is mostly just changing where the name is specified, but it means that one less line of this file needs to be changed. It also, but this doesn't matter, calculates the name at compile time rather than run time.